### ISSUE TITLE
'no_proxy' Support Not Working

### ISSUE THREAD

**Issue by Jakauppila** [adamsb6/s3_file#87]
It doesn't appear that the 'no_proxy' configuration is working correctly.

My test machine has the 'http_proxy', 'https_proxy', and 'no_proxy' environment variables defined and when I attempt to utilize s3_file to pull a file down, I get the following error:

`[2016-09-21T09:06:48-07:00] FATAL: URI::InvalidURIError: javabat[Install JavaBat] (javabat::default line 10) had an error: URI::InvalidURIError: s3_file[C:\Users\vagrant\AppData\Local\Temp\kitchen\cache/javabatch-1.1.0.zip)] (C:\Users\vagrant\AppData\Local\Temp\kitchen\cookbooks\javabat\resources\javabat.rb line 40) had an error: URI::InvalidURIError: bad URI(is not URI?): localhost, 127.0.0.1`

```
libraries/s3_file.rb

def self.client
  require 'rest-client'
  RestClient.proxy = ENV['http_proxy']
  RestClient.proxy = ENV['https_proxy']
  RestClient.proxy = ENV['no_proxy']
  RestClient
end
```
Not sure if it's an issue with how it's passed to rest-client or within rest-client.

**Comment by joekiller** [comment]
Looks like RestClient didn't support this until 2.0: https://github.com/rest-client/rest-client/commit/d2ceac62cf86c3b1418c598ad9148cc574054cab

Try bumping the version in the [attributes.rb](https://github.com/adamsb6/s3_file/blob/master/attributes/default.rb#L2)

**Comment by Jakauppila** [comment]
I can't find which version of Ruby specifically added the native reading of the 'http_proxy' environment variable within 'net/http'.

The commit you reference above simply allows the user to explicitly disable the proxy on a request whereas prior to that, you had to override the ENV['http_proxy'] value to do so.

It doesn't appear that rest-client (or net/http for that matter) actually supports 'no_proxy', so the addition of ``RestClient.proxy = ENV['no_proxy']`` isn't valid, since that would be setting the proxy value to be an array of domain extensions that should bypass the proxy. http://www.gnu.org/software/wget/manual/html_node/Proxies.html

Here's a bug filed to add it as a parameter, https://bugs.ruby-lang.org/issues/11195

But for now, we should just remove ``RestClient.proxy = ENV['no_proxy']``.

**Comment by joekiller** [comment]
closed via #90

**Comment by Jakauppila** [comment]
Any particular reason #90 was merged and not #88?

**Comment by joekiller** [comment]
It added "Add .chefignore so cookbook can be frozen."
